lib/Net/SAML2/Binding/SOAP.pm

```diff
@@ -176,17 +222,51 @@ sub handle_request {
 
     my $saml = _get_saml_from_soap($request);
     if (defined $saml) {
-        $self->verify_xml(
-            $saml,
-            cert_text => $self->idp_cert,
-            cacert    => $self->cacert
-        );
+        foreach my $cert (@{$self->idp_cert}) {
+            $self->verify_xml(
+                $saml,
+                cert_text => $cert,
+                cacert    => $self->cacert
+            );
+        }
         return $saml;
     }
 
     return;
 }
 
+=head2 extract_artifact_message( $response, $message )
```

Comment:
I would prefer to call this from the actual handle_response as a private function.  The current Assertion code works though for finding the Assertion in samlp:ArtifactResponse.  I suspect no one really cares if we take the assertion out of the  samlp:ArtifactResponse tags.

Reply:
Let's discuss this on irc for a moment.